textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/source/IndexSearchSourceOperator.java

```diff
@@ -2,27 +2,47 @@
 
 import edu.uci.ics.textdb.api.common.ITuple;
 import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
 
 /**
  * Created by chenli on 3/28/16.
  */
 public class IndexSearchSourceOperator implements ISourceOperator {
 
+private IDataReader dataReader;
+    
+    public IndexSearchSourceOperator(IDataReader dataReader) throws DataFlowException {
+        this.dataReader = dataReader;
+    }
+
     @Override
-    public void open() {
-        // TODO Auto-generated method stub
-        
+    public void open() throws DataFlowException {
+        try {
+            dataReader.open();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
     }
 
     @Override
-    public ITuple getNextTuple() {
-        // TODO Auto-generated method stub
-        return null;
+    public ITuple getNextTuple() throws DataFlowException {
+        try {
+            return dataReader.getNextTuple();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
     }
 
     @Override
-    public void close() {
-        // TODO Auto-generated method stub
-        
+    public void close() throws DataFlowException {
+        try {
+            dataReader.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
     }

```

Comment:
These implementations of open and close seem pretty standard in everyone's operators. Is it maybe possible to modularize it and inherit it? @JavierJia @chenlica